src/c/backend/scheduler.cpp

```diff
@@ -288,16 +302,7 @@ void InnerScheduler::task_cleanup(InnerWorker *worker, InnerTask *task,
   worker->remove_task();
   this->resources->increase(task->resources);
   this->enqueue_worker(worker);
-
-  // NOTE: Task::complete is NOT equivalent to task->complete
-  // Task->complete:
-  //      - only signals that notify_dependencies has been completed
-  //      - Has a valid state for use in dependency handling
-
-  // Task::complete:
-  //     - signals that the task has finished everything
-
-  task->set_state(state);
+  // task->set_state(state);
```

Comment:
When does this task become 'Completed'? If this happens at anywhere, I am fine though.

Reply:
.. it doesn't (since I only ever needed the `dispatch` flag which still needs to be renamed). I'll fix this before merging.